### PR TITLE
Allow for settings to be partially updated

### PR DESCRIPTION
**Pull request by peterjan**
This PR enables partially updating settings through `PATCH` routes.

Related to https://github.com/SiaFoundation/renterd/issues/1496

### CODE REVIEW

File: bus/bus.go [hunk 1]
```diff
@@ -649,6 +655,59 @@ func (b *Bus) broadcastContract(ctx context.Context, fcid types.FileContractID)
 	return
 }
 
+func (b *Bus) fetchSetting(ctx context.Context, key string) (interface{}, error) {
```
Comment (peterjan) [inline bus/bus.go:658]: This could move to a settings manager 👀

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
         uses: SiaFoundation/action-golang-analysis@HEAD
         with:
           analyzers: |
-            go.sia.tech/jape.Analyzer@master
+            go.sia.tech/jape.Analyzer@d87dee9
```
Comment (peterjan) [inline .github/workflows/test.yml:23]: reminder to revert this back to `master`

File: bus/bus.go (hunk 1 above)
Comment (n8maninger) [reply bus/bus.go:658]: This reduces type safety and hides the implementation without significant benefit imo. Having individual methods like `fetchGougingSettingsOrDefault` would be my preference.

File: bus/bus.go [hunk 3]
```diff
@@ -697,6 +756,74 @@ func (b *Bus) isPassedV2AllowHeight() bool {
 	return cs.Index.Height >= cs.Network.HardforkV2.AllowHeight
 }
 
+func (b *Bus) patchSetting(ctx context.Context, key string, patch map[string]any) (interface{}, error) {
```
Comment (n8maninger) [inline bus/bus.go:759]: nit: prefer `any`

**Review by n8maninger — COMMENTED** [review]
My suggestion would be to break fetch, update, and patch into type safe methods for each setting group. Otherwise, define consts for all the magic strings.

**Comment by peterjan** [comment]
> My suggestion would be to break fetch and update into type safe methods for each setting. Otherwise, define consts for all the magic strings.

went for latter because we already had them defined

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**